great_expectations/datasource/new_datasource.py

```diff
@@ -59,14 +61,16 @@ def __init__(
                 runtime_environment={"concurrency": concurrency},
                 config_defaults={"module_name": "great_expectations.execution_engine"},
             )
-            self._datasource_config: dict = {
-                "execution_engine": execution_engine,
-                "id": id,
-                "name": name,
-            }
         except Exception as e:
             raise ge_exceptions.ExecutionEngineError(message=str(e))
 
+        self._datasource_config: dict = {
+            "execution_engine": execution_engine,
+            "id": id,
+            "name": name,
+        }
+        self._raw_config = raw_config or self._datasource_config
```

Comment:
In the case a user instantiates a `Datasource` outside our desired context flow, we need to ensure that their `raw_config` is populated if they want to persist.

Example:
```python
datasource = Datasource(...)
context.save_datasource(datasource)
```

What if I include creds in my naked datasource? Do we allow users to leak values or should be raise warnings?

Reply:
This is a good question. I'm not sure on this one - would be good to raise with the team.